youtube_dl/extractor/xiami.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Created by i@BlahGeek.com at 2016-04-04
+
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import xpath_element, xpath_text, ExtractorError
+import re
+
+try:
+    from urllib import unquote
+except ImportError:
+    from urllib.parse import unquote
+
+
+class XiamiIE(InfoExtractor):
+    IE_NAME = 'xiami'
+    IE_DESC = '虾米音乐'
+    _VALID_URL = r'http://www\.xiami\.com/(?P<typ>(\w+))/(?P<id>[0-9]+)'
+
+    _TESTS = [
+        {
+            'url': 'http://www.xiami.com/song/1775610518',
+            'md5': '521dd6bea40fd5c9c69f913c232cb57e',
+            'info_dict': {
+                'id': '1775610518',
+                'ext': 'mp3',
+                'title': 'Woman',
+                'creator': 'HONNE',
+                'album': 'Woman',
+                'thumbnail': r're:http://img\.xiami\.net/images/album/.*\.jpg',
+                'description': 'md5:052ec7de41ca19f67e7fd70a1bfc4e0b',
+            }
+        },
+        {
+            'url': 'http://www.xiami.com/song/1775256504',
+            'md5': '932a3abd45c6aa2b1fdbe028fcb4c4fc',
+            'info_dict': {
+                'id': '1775256504',
+                'ext': 'mp3',
+                'title': '悟空',
+                'creator': '戴荃',
+                'album': '悟空',
+                'description': 'md5:206e67e84f9bed1d473d04196a00b990',
+            }
+        },
+        {
+            'url': 'http://www.xiami.com/album/2100300444',
+            'info_dict': {
+                'id': '2100300444',
+            },
+            'playlist_count': 10,
+        },
+        {
+            'url': 'http://www.xiami.com/album/512288?spm=a1z1s.6843761.1110925389.6.hhE9p9',
+            'only_matching': True,
+        }
+    ]
+
+    _TYPES = {
+        'song': '',
+        'album': '/type/1',
+        'artist': '/type/2',
+        'collect': '/type/3',
+    }
+
+    _XML_BASE_URL = 'http://www.xiami.com/song/playlist/id'
+    _XML_NS = '{http://xspf.org/ns/0/}'
+
+    def _real_extract(self, url):
+        match = re.match(self._VALID_URL, url)
+        assert(match)
+        typ, _id = match.group('typ'), match.group('id')
+
+        if typ not in self._TYPES:
+            raise ExtractorError('URL not supported')
+        xml_url = '%s/%s' % (self._XML_BASE_URL, _id)
+        xml_url += self._TYPES[typ]
+
+        playlist = self._download_xml(xml_url, _id)
+        tracklist = xpath_element(playlist, './' + self._XML_NS + 'trackList')
+
+        def _extract_single(track):
+            artist = xpath_text(track, self._XML_NS + 'artist', default='')
+            artist = artist.split(';')
+
+            ret = {
+                'id': xpath_text(track, self._XML_NS + 'song_id'),
+                'title': xpath_text(track, self._XML_NS + 'title'),
+                'album': xpath_text(track, self._XML_NS + 'album_name'),
+                'artist': ';'.join(artist) if artist else None,
```

Comment:
@dstftw about that, I mentioned it in #9076 , fields like `album` and `artist` is not in the documentation of InfoExtractor, but FFmpegMetadataPP uses these fields as metadata. So I think it's worth including these fields here for better metadata handling.